### PR TITLE
Migration to 1ES Templates

### PR DESCRIPTION
**Pull request by payton2022**
I have migrated the Microsoft SARIF website's code repo to 1ES compliance, the original code base can be seen [here](https://github.com/microsoft/sarif-website). A successful run of the new pipeline can be seen [here](https://dev.azure.com/mseng/1ES/_build/results?buildId=18235559&view=results). The original pipeline before I made these changes can be seen [here](https://dev.azure.com/mseng/1ES/_build/results?buildId=18235559&view=results).

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -4,11 +4,38 @@ pr:
...
+            inputs:
+              packageType: sdk
+              version: 5.0.x
+          - task: DotNetCoreCLI@2
+            inputs:
+              command: 'build'
+              projects: src/SarifWeb.sln
+              configuration: 'Release'
+
+          - task: ComponentGovernanceComponentDetection@0
```
Comment (sirkiza) [inline azure-pipelines.yml:41]: Is this task required? Check if 1ES perform that validation

Comment (payton2022) [reply azure-pipelines.yml]: removed, as this is not required

File: azure-pipelines.yml [hunk 2]
```diff
@@ -4,11 +4,46 @@ pr:
...
+        
+      stages:
+      - stage: build
+        jobs:
+          - job: build
+            templateContext:
+              inputs:
+              - input: checkout
+                repository: self
+                displayName: build
```
Comment (sirkiza) [inline azure-pipelines.yml:34]: Why do we need `displayName` here?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -4,11 +4,46 @@ pr:
...
+          tenantId: PME
+          identityType: 1ESPipelineIdentity
+        
+      stages:
+      - stage: build
+        jobs:
+          - job: build
+            templateContext:
+              inputs:
+              - input: checkout
```
Comment (sirkiza) [inline azure-pipelines.yml:32]: in general, this `input` seems redundant

File: azure-pipelines.yml [hunk 4]
```diff
@@ -4,11 +4,46 @@ pr:
...
+                repository: self
+                displayName: build
+              outputs:
+              - output: pipelineArtifact
+                targetPath: $(System.DefaultWorkingDirectory)
 
-- task: ComponentGovernanceComponentDetection@0
+            steps:
+              - task: UseDotNet@2
```
Comment (sirkiza) [inline azure-pipelines.yml:40]: you are using container that should have dotnet. it doesn't seem right to install it one more time

File: azure-pipelines.yml (hunk 2 above)
Comment (payton2022) [reply azure-pipelines.yml]: removed

File: azure-pipelines.yml [hunk 5]
```diff
@@ -4,11 +4,46 @@ pr:
...
+      stages:
+      - stage: build
+        jobs:
+          - job: build
+            templateContext:
+              inputs:
+              - input: checkout
+                repository: self
+                displayName: build
+              outputs:
```
[the remark this answers is not in the document]
Comment (payton2022) [reply azure-pipelines.yml:34]: added an artifact name

File: azure-pipelines.yml (hunk 3 above)
Comment (payton2022) [reply azure-pipelines.yml]: removed input

File: azure-pipelines.yml (hunk 4 above)
Comment (payton2022) [reply azure-pipelines.yml]: removed

**Review by EasyRhinoMSFT — APPROVED** [review]
:shipit:

**Pull request merged**